src/mono/wasm/debugger/BrowserDebugProxy/ExpressionEvaluator.cs

```diff
@@ -166,6 +167,8 @@ public SyntaxTree ReplaceVars(SyntaxTree syntaxTree, IEnumerable<JObject> ma_val
                         memberAccessValues[id_name] = value;
                         AddLocalVariableWithValue(id_name, value);
                     }
+                    // do not replace memberAccesses that were already replaced
+                    memberAccesses = new List<MemberAccessExpressionSyntax>();
```

Comment:
Why exactly is this needed?

Reply:
Let's assume we not emptying the list and resolving: `test.propString.Split('*', 3, System.StringSplitOptions.RemoveEmptyEntries)`.
Then here: https://github.com/dotnet/runtime/blob/8e6ab78d84c0053da776f0cb04c7b9f029297301/src/mono/wasm/debugger/BrowserDebugProxy/ExpressionEvaluator.cs#L398 we are resolving System.StringSplitOptions.RemoveEmptyEntries. Then, as it's a method call we are entering the `ReplaceVars` method again, now only with `newMemberAccessValues` that has one element: `test.propString` but replacer.memberAccesses have already 2 elements (0: enum, 1: string), so zip will pair them wrong, enum will get the string value and string will be skipped as it does not have a matching value in `ma_values`. That's why we have to clear `memberAccesses` after the first call. 
It should work with `replacer.identifiers` the same way but I could not find a test case to break the current version of code, so I did not add the clean-up there.

Reply:
The best way would be to replace the Lists of SyntaxElement in Replacer with <Key, Value> collections. It would simplify the logic much more and make the code more legible (e.g. no nulls passed into ReplaceVars, just Replacer instance) but it is a target for some separate PR.